tests/test_cases/socketselect.c

```diff
@@ -0,0 +1,331 @@
+
+   /* Initialize the timeval struct to 3 minutes.  If no        */
+   /* activity after 3 minutes this program will end.           */
+   
+   timeout.tv_sec  = 5;
+   timeout.tv_usec = 0;
+
+   
+
+   
+   /* Loop waiting for incoming connects or for incoming data   */
+   /* on any of the connected sockets.                          */
+   
+   do
+   {
+      /* Copy the master fd_set over to the working fd_set.     */
+      memcpy(&working_set, &master_set, sizeof(master_set));
+
+      /* Call select() and wait for it to complete.   */
+      printf("Waiting on select()...\n");
+      fflush(stdout);
+      rc = select(max_sd + 1, &working_set, NULL, NULL, &timeout);
+
+      /* Check to see if the select call failed.                */
+      if (rc < 0)
+      {
+         perror("  select() failed");
+         break;
+      }
+
+      /* Check to see if the time out expired.         */
+      if (rc == 0)
+      {
+         printf("  select() timed out.  End program.\n");
+         break;
+      }
+
+      /* One or more descriptors are readable.  Need to         */
+      /* determine which ones they are.                         */
+      desc_ready = rc;
+      for (i=0; i <= max_sd  &&  desc_ready > 0; ++i)
+      {
+         /* Check to see if this descriptor is ready            */
+         if (FD_ISSET(i, &working_set))
+         {
+            
+            /* A descriptor was found that was readable - one   */
+            /* less has to be looked for.  This is being done   */
+            /* so that we can stop looking at the working set   */
+            /* once we have found all of the descriptors that   */
+            /* were ready.                                      */
+            
+            desc_ready -= 1;
+        
+            /* Check to see if this is the listening socket     */
+            
+            if (i == listen_sd)
+            {
+               printf("  Listening socket is readable\n");
+               fflush(stdout);
+               
+               /* Accept all incoming connections that are      */
+               /* queued up on the listening socket before we   */
+               /* loop back and call select again.              */
+               
+               do
+               {
+                  
+                  /* Accept each incoming connection.  If       */
+                  /* accept fails with EWOULDBLOCK, then we     */
+                  /* have accepted all of them.  Any other      */
+                  /* failure on accept will cause us to end the */
+                  /* server.                                    */
+                  
+                  new_sd = accept(listen_sd, NULL, NULL);
+                  if (new_sd < 0)
+                  {
+                    //  if (errno != EWOULDBLOCK)
+                    //  {
+                    //     perror("  accept() failed");
+                    //     end_server = TRUE;
+                    //  }
```

Comment:
Why is this commented out?

Reply:
Ah this was commented out when I was trying to get this to work despite the errnos being out of whack. Can uncomment this now.